jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/SamDebugger.kt

```diff
@@ -28,12 +36,37 @@ internal class SamDebugger : SamRunner() {
             .patchCommandLine(debugPort, state, commandLine)
     }
 
-    override fun run(environment: ExecutionEnvironment, state: SamRunningState): RunContentDescriptor {
+    override fun run(environment: ExecutionEnvironment, state: SamRunningState): Promise<RunContentDescriptor> {
         val debugSupport = SamDebugSupport.getInstanceOrThrow(state.settings.runtimeGroup)
-        val debugProcess = debugSupport.createDebugProcess(environment, state, debugPort)
-        val debugManager = XDebuggerManager.getInstance(environment.project)
-        return debugProcess?.let {
-            return debugManager.startSession(environment, debugProcess).runContentDescriptor
-        } ?: throw IllegalStateException("Failed to create debug process")
+        val promise = AsyncPromise<RunContentDescriptor>()
+
+        var isDebuggerAttachDone = false
+
+        ProgressManager.getInstance().run(object : Task.Backgroundable(environment.project, message("lambda.debug.waiting"), false) {
+            override fun run(indicator: ProgressIndicator) {
+                spinUntil(debugSupport.debuggerAttachTimeoutMs) { isDebuggerAttachDone }
```

Comment:
Here you can check return value of spinUnit. In case of 'false' maybe it worth to log a problem

Reply:
Fixed.